src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -11306,6 +11306,29 @@ instruct convD2L_reg_reg(rRegL dst, regD src, rFlagsReg cr)
   ins_pipe(pipe_slow);
 %}
 
+instruct convF2HF_reg_reg(rRegI dst, regF src, regF tmp) %{
+  effect(TEMP tmp);
+  match(Set dst (ConvF2HF src));
+  format %{ "vcvtps2ph $dst,$src \t using $tmp as TEMP"%}
+  ins_encode %{
+     __ vcvtps2ph($tmp$$XMMRegister, $src$$XMMRegister, 0x04, Assembler::AVX_128bit);
+     __ movdl($dst$$Register, $tmp$$XMMRegister);
+     __ movswl($dst$$Register, $dst$$Register);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
```

Comment:
For F2HF, good to also add optimized rule with StoreC to benefit from vcvtps2ph memory destination form of instruction.
match(Set mem (StoreC mem (ConvF2HF src)));